android/src/main/java/com/robinpowered/RNMDMManager/RNMobileDeviceManagerModule.java

```diff
@@ -39,14 +43,67 @@ public void onReceive(Context context, Intent intent) {
                 }
                 if (thisContext.hasActiveCatalystInstance()) {
                     thisContext
-                        .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
-                        .emit(APP_CONFIG_CHANGED, data);
+                            .getJSModule(DeviceEventManagerModule.RCTDeviceEventEmitter.class)
+                            .emit(APP_CONFIG_CHANGED, data);
                 }
             }
         };
         thisContext.registerReceiver(restrictionReceiver,restrictionFilter);
     }
 
+    public boolean enableLockState() {
+        if (!isLockState()) {
+            Activity activity = getCurrentActivity();
+            if (activity == null) {
+                return false;
+            }
+            activity.startLockTask();
+            return true;
+        }
+        return false;
+    }
+
+    public boolean disableLockState() {
+        if (isLockState()) {
+            try {
+                Activity activity = getCurrentActivity();
+                if (activity == null) {
+                    return false;
+                }
+                activity.stopLockTask();
+                return true;
+            } catch (SecurityException e) {
+                return false;
+            }
+        }
+        return false;
+    }
+
+    public boolean isLockStatePermitted() {
+        DevicePolicyManager dpm = (DevicePolicyManager)
+                getReactApplicationContext().getSystemService(Context.DEVICE_POLICY_SERVICE);
+        return dpm.isLockTaskPermitted(getReactApplicationContext().getPackageName());
+    }
+
+    public boolean isLockState() {
+        boolean isLocked = false;
+        ActivityManager am = (ActivityManager) getReactApplicationContext().getSystemService(Context.ACTIVITY_SERVICE);
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+                if (am.getLockTaskModeState() != ActivityManager.LOCK_TASK_MODE_NONE) {
+                    isLocked = true;
+                }
+            } else {
+                if (am.isInLockTaskMode()) {
+                    isLocked = true;
+                }
+            }
+        } catch (e) {
```

Comment:
Looking at the Android source code, seems like it will throw a RemoteException.

```
public int getLockTaskModeState() {
  try {
    return getService().getLockTaskModeState();
  } catch (RemoteException e) {
    throw e.rethrowFromSystemServer();
  }
}
```

Reply:
https://android.googlesource.com/platform/frameworks/base.git/+/master/core/java/android/app/ActivityManager.java#4346

Reply:
I think it would be reasonable to turn this into a promise rejection